rustainers/src/runner/inner.rs

```diff
@@ -185,11 +189,48 @@ pub(crate) trait InnerRunner: Display + Debug + Send + Sync {
         &self,
         id: ContainerId,
         network: &str,
-    ) -> Result<ContainerNetwork, ContainerError> {
-        let path = format!(".NetworkSettings.Networks.{network}");
+    ) -> Result<NetworkDetails, ContainerError> {
+        let mut networks = self.inspect_networks(id).await?;
+        if let Some((_, network)) = networks.remove_entry(network) {
+            Ok(network)
+        } else {
+            Err(ContainerError::NoNetwork)
+        }
+    }
+
+    #[tracing::instrument(level = "debug", skip(self), fields(runner = %self))]
+    async fn inspect_networks(
+        &self,
+        id: ContainerId,
+    ) -> Result<HashMap<String, NetworkDetails>, ContainerError> {
+        let path = ".NetworkSettings.Networks".to_string();
+        self.inspect(id, &path).await
+    }
+
+    #[tracing::instrument(level = "debug", skip(self), fields(runner = %self))]
+    async fn inspect_host_containers(
+        &self,
+        id: ContainerId,
+    ) -> Result<HashMap<ContainerId, HostContainer>, ContainerError> {
+        let path = ".Containers".to_string();
         self.inspect(id, &path).await
     }
 
+    #[tracing::instrument(level = "debug", skip(self), fields(runner = %self))]
+    async fn list_networks(&self, name: &str) -> Result<Vec<NetworkInfo>, ContainerError> {
```

Comment:
🎯 **suggestion**: I think that we can update that method to be compatible with podman.

* remove the `name` parameter
* remove the `--filter`and `&format!("type={name}")` arguments
* remove networks with name in ['bridge', 'host', 'none'] with the `retain` vec method
* rename `list_custom_networks`

Reply:
I don't think that will work. Indeed according to [doc](https://docs.docker.com/reference/cli/docker/network/ls/#type) a custom network type refers to a one created by the user.

The network in a docker is a custom and a bridge network. I would need to find a better filtering

Reply:
Look at the diff

```
❯ docker network ls
NETWORK ID     NAME                                                         DRIVER    SCOPE
93356d7ec423   bridge                                                       bridge    local
559a7c181eda   host                                                         host      local
69bd6a66eb9d   mochoa_pgadmin4-docker-extension-desktop-extension_default   bridge    local
c68252520931   my_network_01HPBT7KXXB0FPZZQMKBFV2WYN                        bridge    local
e604b1c7e718   my_network_01HPBT7KXXFWD45G7ZHJ1R7WZC                        bridge    local
7187677565db   my_network_01HPBT7KXXZTXSMDESXT8MMCAK                        bridge    local
6bc3645c5129   my_network_01HPBTCAKTC6T71RVJC5M3SAZN                        bridge    local
5c54518de27a   my_network_01HPBTCAKTDXNSWNH52XB3YZF9                        bridge    local
938de663ef58   my_network_01HPBTCAKTEJ7Z448FFHAR35PN                        bridge    local
9007a1874e10   my_network_01HPBTGBTRYQ47VGJHG1JFM7KS                        bridge    local
24a90a5d256b   my_network_01HPBTGBTSKK1C4K7ASGJA3GKK                        bridge    local
54cd782d3ce0   my_network_01HPBTGBTSYGKHBTRJ8BT5C3QK                        bridge    local
55be55f0887c   my_network_01HR1RK3SP82341KV1HYF5HCEN                        bridge    local
9810b3a02a33   my_network_01HR1RK3SQ0BJRK9QD00ER0DSV                        bridge    local
d8acd826c65f   my_network_01HR1RK3Y1A8BPR5GP73TBS3K7                        bridge    local
44f627932856   none                                                         null      local

❯ docker network ls --filter type=custom
NETWORK ID     NAME                                                         DRIVER    SCOPE
69bd6a66eb9d   mochoa_pgadmin4-docker-extension-desktop-extension_default   bridge    local
c68252520931   my_network_01HPBT7KXXB0FPZZQMKBFV2WYN                        bridge    local
e604b1c7e718   my_network_01HPBT7KXXFWD45G7ZHJ1R7WZC                        bridge    local
7187677565db   my_network_01HPBT7KXXZTXSMDESXT8MMCAK                        bridge    local
6bc3645c5129   my_network_01HPBTCAKTC6T71RVJC5M3SAZN                        bridge    local
5c54518de27a   my_network_01HPBTCAKTDXNSWNH52XB3YZF9                        bridge    local
938de663ef58   my_network_01HPBTCAKTEJ7Z448FFHAR35PN                        bridge    local
9007a1874e10   my_network_01HPBTGBTRYQ47VGJHG1JFM7KS                        bridge    local
24a90a5d256b   my_network_01HPBTGBTSKK1C4K7ASGJA3GKK                        bridge    local
54cd782d3ce0   my_network_01HPBTGBTSYGKHBTRJ8BT5C3QK                        bridge    local
55be55f0887c   my_network_01HR1RK3SP82341KV1HYF5HCEN                        bridge    local
9810b3a02a33   my_network_01HR1RK3SQ0BJRK9QD00ER0DSV                        bridge    local
d8acd826c65f   my_network_01HR1RK3Y1A8BPR5GP73TBS3K7                        bridge    local
```

Reply:
Inside a docker
```
➜ docker run  -it --rm --name dockerindocker -v $(pwd):/workdir -v /var/run/docker.sock:/var/run/docker.sock --cpus 5 --memory 10g --network dummy_network d_in_d
root@85f57161e149:/# docker network ls
NETWORK ID     NAME            DRIVER    SCOPE
ed4d23711a70   bridge          bridge    local
d00f9f486b84   dummy_network   bridge    local
bc2f8d4c9325   host            host      local
571b884aad60   none            null      local
root@85f57161e149:/# docker network ls --filter type=custom
NETWORK ID     NAME            DRIVER    SCOPE
d00f9f486b84   dummy_network   bridge    local
```

Reply:
So you just need to remove networks with **name** 'bridge', 'host', 'none' to be equivalent to the `filter=custom`